crates/rust-analyzer/src/bin/args.rs

```diff
@@ -47,12 +48,25 @@ pub(crate) enum Command {
 }
 
 impl Args {
+    fn get_roots(mut matches: Arguments) -> Result<Option<Vec<String>>> {
+        if matches.contains("--roots") {
```

Comment:
Adding a command line option, given that this is the first command line option, massively changes the interface of the program. It's much better to stick to existing conventions, which, in this case, are the initializationOptions (not sure about the naming) field of the initialize requst

Reply:
> massively changes the interface of the program

As I said I couldn't find a way to send it via the initialization options in vscode. It also shouldn't be a massive change to the interface since it's a completely optional parameter but I can see how it feels intrusive.